xblockutils/studio_editable.py

```diff
@@ -0,0 +1,320 @@
+                "not generally created/configured by content authors in Studio."
+            )
+            field_info = self._make_field_info(field_name, field)
+            if field_info is not None:
+                context["fields"].append(field_info)
+        fragment.content = loader.render_template('templates/studio_edit.html', context)
+        fragment.add_javascript(loader.load_unicode('public/studio_edit.js'))
+        fragment.initialize_js('StudioEditableXBlockMixin')
+        return fragment
+
+    def _make_field_info(self, field_name, field):
+        """
+        Create the information that the template needs to render a form field for this field.
+        """
+        supported_field_types = (
+            (Integer, 'integer'),
+            (Float, 'float'),
+            (Boolean, 'boolean'),
+            (String, 'string'),
+            (List, 'list'),
+            (JSONField, 'generic'),  # This is last so as a last resort we display a text field w/ the JSON string
+        )
+        info = {
+            'name': field_name,
+            'display_name': field.display_name,
+            'is_set': field.is_set_on(self),
+            'default': field.default,
+            'value': field.read_from(self),
+            'has_values': False,
+            'help': field.help,
+            'allow_reset': field.runtime_options.get('resettable_editor', True),
+            'list_values': None,  # Only available for List fields
+            'has_list_values': False,  # True if list_values_provider exists, even if it returned no available options
+        }
+        for type_class, type_name in supported_field_types:
+            if isinstance(field, type_class):
+                info['type'] = type_name
+                # If String fields are declared like String(..., multiline_editor=True), then call them "text" type:
+                editor_type = field.runtime_options.get('multiline_editor')
+                if type_class is String and editor_type:
+                    if editor_type == "html":
+                        info['type'] = 'html'
+                    else:
+                        info['type'] = 'text'
+                if type_class is List and field.runtime_options.get('list_style') == "set":
+                    # List represents unordered, unique items, optionally drawn from list_values_provider()
+                    info['type'] = 'set'
+                elif type_class is List:
+                    info['type'] = "generic"  # disable other types of list for now until properly implemented
+                break
+        if "type" not in info:
+            raise NotImplementedError("StudioEditableXBlockMixin currently only supports fields derived from JSONField")
+        if info["type"] in ("list", "set"):
+            info["value"] = [json.dumps(val) for val in info["value"]]
+            if info["default"] is None:
+                info["default"] = []
```

Comment:
Diff-cover reports _this_ line - and a few others- are untested.
```
xblockutils.studio_editable     145     29    80%   137, 141, 149, 158-162, 201, 255-257, 272-286, 298-304, 311-313, 320
```

I don't think 100% coverage is required, just want to make sure you've looked over the list of untested lines and made a judgement for each instance as to whether or not it's worth adding an additional test.